app/assets/javascripts/discourse/app/services/route-history.js

```diff
@@ -0,0 +1,36 @@
+import { tracked } from "@glimmer/tracking";
+import Service, { service } from "@ember/service";
+import { bind } from "discourse/lib/decorators";
+import { defaultHomepage } from "discourse/lib/utilities";
+
+// This service is responsible for managing the route history
+// mainly used by the `BackButton` component
+export default class RouteHistory extends Service {
+  @service router;
+
+  @tracked lastUrl = null;
```

Comment:
This state will be lost whenever the page is refreshed, which happens whenever a code update is deployed to the forum. Maybe we can try storing it in sessionStorage instead?

(see internal `/t/151654/7`)